coding-standards-and-naming.rst

```diff
@@ -264,17 +265,16 @@ library. The application of common sense is recommended.
    constant to specify different behavior or actions SHOULD be done through a
    ``#define``.
 
-#. When writing functions that deal with strings, be sure to remember that PHP
-   holds the length property of each string, and that it shouldn't be calculated
-   with ``strlen()``. Write your functions in such a way so that they'll take
-   advantage of the length property, both for efficiency and in order for them
-   to be binary-safe. Functions that change strings and obtain their new lengths
-   while doing so, should return that new length, so it doesn't have to be
-   recalculated with ``strlen()`` (e.g. ``php_addslashes()``).
+#. When writing functions that deal with strings, you SHOULD use
+   ``zend_string``, which holds the value and the length property of each
+   string.
 
-#. You MUST NOT use ``strncat()``. If you're absolutely sure you know what
-   you're doing, check its man page again, and only then, consider using it, and
-   even then, try avoiding it.
+   Write your functions in such a way so that they'll take advantage of the
+   length property by using ``ZSTR_LEN()``, both for efficiency, and in order
+   for them to be binary-safe.
+
+#. You SHOULD use the ``smart_str_*`` family of functions for string creation,
+   instead of relying on the C-library versions, such as ``strncat()``.
```

Comment:
smart_str_* should be used only if the string is constructed with multiple variable pieces. Otherwise basic creation should be done using zend_string_alloc and similar (e.g. short concatenations using zend_string_concat2 and zend_string_concat3).

That said I'm not really sure why this is in coding standard. This should really be part of some extension documentation like in doc/ or php internals book.